include/cvc5/cvc5_proof_rule.h

```diff
@@ -2452,6 +2452,21 @@ enum ENUM(ProofRewriteRule)
    * \endverbatim
    */
   EVALUE(QUANT_MINISCOPE),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Quantifiers -- Miniscoping free variables**
+   *
+   * .. math::
+   *   \forall X.\> F_1 \vee \ldots \vee F_n =
+   *   (\forall X_1.\> F_1) \vee \ldots \vee (\forall X_n.\> F_n)
+   * 
+   * where each variable in :math:`X` occurs in at most one of
+   * :math:`X_1 \ldots X_n`, and the right hand side does not have any free
+   * variable in :math:`X`.
```

Comment:
Could we change `at most one of` to `exactly one of`? We could call `QUANT_UNUSED_VARS` once before this rule to get rid of all unused variables. Also, how is $$X$$ related to $$X_1 \ldots X_n$$? My preference is $$X = X_1 \text{++} \cdots \text{++} X_n$$. As in: $$X$$ is the concatenation of $$X_1$$ through $$X_n$$. If we need the flexibility, we could add another rewrite rule: $$(\forall X. F) = (\forall Y. F)$$ where $$Y$$ is a reordering of $$X$$. This would only be called once too!

Reply:
What is the context here in which the reordering is needed? Depending on what we need to model, a rule that transforms `forall xy. P --> forall yx. P` is just some kind of prenexing.

Reply:
Here is an example that requires reordering based on the form of the rule Abdal wants:
```
forall xyz. (or (P x z) (Q y)) ---> (forall xzy. (or (P x z) (Q y)) ----> (or (forall xz. (P x z)) (forall y. (Q y))
```
The second rewrite is "easier" if the variables are ordered. This requires reordering them in a first step.

Reply:
Thinking about this some, I actually will need this reordering rule for the reconstruction of two other forthcoming rules.

I would advocate we have this rule in its current form, although I am open to suggestions on the name (currently QUANT_VAR_REORDERING).

Reply:
Sounds good